modules/custom/advanced_help_block/src/Form/AdvancedHelpBlockSettingsForm.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\advanced_help_block\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Class AdvancedHelpBlockSettingsForm.
+ *
+ * @package Drupal\advanced_help_block\Form
+ * @ingroup advanced_help_block
+ */
+class AdvancedHelpBlockSettingsForm extends FormBase {
+  /**
+   * Returns a unique string identifying the form.
+   *
+   * @return string
+   *   The unique string identifying the form.
+   */
+  public function getFormId() {
+    return 'advanced_help_block_settings';
+  }
+
+  /**
+   * Form submission handler.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param FormStateInterface $form_state
+   *   An associative array containing the current state of the form.
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $config_factory = \Drupal::configFactory();
+    $config_factory->getEditable('advanced_help_block.settings')
+      ->set('advanced_help_block.view_type', $form_state->getValue('view_type'))
+      ->save();
+    drupal_set_message(t('Advanced help block type was changed to <b>@value</b>', ['@value' => $form_state->getValue('view_type')]));
+  }
+
+
+  /**
+   * Define the form used for ContentEntityExample settings.
+   *
+   * @return array
+   *   Form definition array.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param FormStateInterface $form_state
+   *   An associative array containing the current state of the form.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config_factory = \Drupal::configFactory();
```

Comment:
Let's get used to Drupal 8 ecosystem

Reply:
changed to  $this->configFactory .... etc